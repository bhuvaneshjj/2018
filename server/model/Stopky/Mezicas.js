'use strict';

const mongoose = require('mongoose');

const MezicasSchema = new mongoose.Schema(
  {
    cas: { type: String, require: true },
    korekce: { type: String },
  },
  { _id: false, bufferCommands: false }
);

module.exports = MezicasSchema;
