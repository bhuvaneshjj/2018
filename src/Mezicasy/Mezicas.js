import React from 'react';
import PropTypes from 'prop-types';
import { Button } from 'react-bootstrap';
import { convertDuration } from '../Util';

const Mezicas = ({ poradi, duration, cislo, onClick }) => {
  let { hours, mins, secs, subsecs } = convertDuration(duration);

  return (
    <tr>
      <td>
        {poradi}
        {'.'}
      </td>
      <td>
        {hours}
        {':'}
        {mins}
        {':'}
        {secs}
        {','}
        {subsecs}
      </td>
      <td>{cislo}</td>
      <td>
        <Button bsStyle="danger" bsSize="xsmall" onClick={onClick}>
          x
        </Button>
      </td>
    </tr>
  );
};

Mezicas.propTypes = {
  poradi: PropTypes.number.isRequired,
  duration: PropTypes.object.isRequired,
  cislo: PropTypes.node.isRequired,
  onClick: PropTypes.func.isRequired
};

export default Mezicas;
