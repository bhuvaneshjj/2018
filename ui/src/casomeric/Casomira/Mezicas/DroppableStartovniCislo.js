import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import { DropTarget } from 'react-dnd';
import { DragTypes } from '../../../constants';
import StartovniCislo from './StartovniCislo';

const collect = (connect, monitor) => ({
  connectDropTarget: connect.dropTarget(),
  isDropOver: monitor.isOver() && monitor.canDrop()
});

class DroppableStartovniCislo extends PureComponent {
  constructor(props) {
    super(props);
    const { canDrop, onDrop, typ } = this.props;

    /* Mixing ...monitor.getItem() and ...dropProps will cause some attributes be overwritten.
       We prefer destination ones over source ones here. */
    this.dropTarget = {
      drop: (dropProps, monitor) => onDrop({ ...monitor.getItem(), ...dropProps, typ }),
      canDrop: (dropProps, monitor) =>
        canDrop({ ...monitor.getItem(), ...dropProps, destinationTyp: typ })
    };
  }

  render = () => {
    const { canDrop, onDrop, ...rest } = this.props;
    const Droppable = DropTarget(DragTypes.STARTOVNI_CISLO, this.dropTarget, collect)(
      StartovniCislo
    );
    return <Droppable {...rest} />;
  };
}

DroppableStartovniCislo.propTypes = {
  typ: PropTypes.string.isRequired,
  canDrop: PropTypes.func.isRequired,
  onDrop: PropTypes.func.isRequired
};

export default DroppableStartovniCislo;
