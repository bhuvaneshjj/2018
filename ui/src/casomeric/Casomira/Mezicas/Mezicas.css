.Mezicas__poradi {
  justify-content: flex-end;
}

.Mezicas__startCislo {
  justify-content: flex-end;
}

.Mezicas__buttons {
  display: flex;
  justify-content: space-between;
}

.Mezicas__button--edit {
  color: blue;
}

.Mezicas__button--remove {
  color: #d38181;
}
