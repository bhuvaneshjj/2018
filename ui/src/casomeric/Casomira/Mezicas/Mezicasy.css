.Mezicasy__grid {
  align-items: stretch;
  display: grid;
  grid-template-columns: 3em 6em 5em 6em;
  justify-content: stretch;
  overflow-y: auto;
}

.Mezicasy__header,
.Mezicasy__leftHeader,
.Mezicasy__middleHeader,
.Mezicasy__cell,
.Mezicasy__leftCell,
.Mezicasy__middleCell {
  align-items: center;
  display: flex;
  padding: 5px;
}

.Mezicasy__leftCell,
.Mezicasy__leftHeader {
  border-left: 1px solid #d3d3d3;
  border-right: 1px solid #d3d3d3;
}
.Mezicasy__middleCell,
.Mezicasy__middleHeader {
  border-right: 1px solid #d3d3d3;
}

.Mezicasy__header,
.Mezicasy__leftHeader,
.Mezicasy__middleHeader {
  background-color: #f6f7f8;
  background: linear-gradient(#ffffff, #efefef);
  border-bottom: 1px solid #d3d3d3;
  font-weight: bold;
  justify-content: center;
}

.Mezicasy__grid > div:nth-child(8n + 1),
.Mezicasy__grid > div:nth-child(8n + 2),
.Mezicasy__grid > div:nth-child(8n + 3),
.Mezicasy__grid > div:nth-child(8n + 4) {
  background-color: #f6f7f8;
}
