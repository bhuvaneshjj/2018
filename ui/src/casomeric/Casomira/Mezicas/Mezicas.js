import React from 'react';
import PropTypes from 'prop-types';
import momentPropTypes from 'react-moment-proptypes';
import { Button, Glyphicon } from 'react-bootstrap';
import { convertDuration } from '../../../Util';
import './Mezicas.css';

const Mezicas = ({ poradi, cas, startCislo, onRemove, onEdit }) => {
  const duration = convertDuration(cas);

  return (
    <React.Fragment>
      <div className="Mezicasy__cell Mezicas__poradi">{poradi}.</div>
      <div className="Mezicasy__leftCell Mezicas__cas">
        {cas && `${duration.hours}:${duration.mins}:${duration.secs},${duration.subsecs}`}
      </div>
      <div className="Mezicasy__middleCell Mezicas__startCislo">{startCislo}</div>
      <div className="Mezicasy__cell Mezicas__buttons">
        <Button bsSize="small" className="Mezicas__button--remove" onClick={onRemove}>
          <Glyphicon glyph="remove" />
        </Button>
        <Button bsSize="small" className="Mezicas__button--edit" onClick={onEdit} disabled={true}>
          <Glyphicon glyph="edit" />
        </Button>
      </div>
    </React.Fragment>
  );
};

Mezicas.propTypes = {
  cas: momentPropTypes.momentDurationObj,
  poradi: PropTypes.number.isRequired,
  startCislo: PropTypes.number,
  onEdit: PropTypes.func.isRequired,
  onRemove: PropTypes.func.isRequired
};

export default Mezicas;
