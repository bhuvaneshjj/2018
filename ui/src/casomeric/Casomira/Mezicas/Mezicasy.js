import React from 'react';
import PropTypes from 'prop-types';
import momentPropTypes from 'react-moment-proptypes';
import Mezicas from './Mezicas';
import './Mezicasy.css';

const Mezicasy = ({ containerHeight, mezicasy, typ }) => (
  <div className="Mezicasy__grid" style={{ height: containerHeight }}>
    <div className="Mezicasy__header">#</div>
    <div className="Mezicasy__leftHeader">čas</div>
    <div className="Mezicasy__middleHeader">číslo</div>
    <div className="Mezicasy__header" />
    {mezicasy.map((mezicas, index) => (
      <Mezicas key={index} poradi={index + 1} {...mezicas} typ={typ} />
    ))}
  </div>
);

Mezicasy.propTypes = {
  containerHeight: PropTypes.number.isRequired,
  mezicasy: PropTypes.arrayOf(
    PropTypes.shape({
      id: PropTypes.string,
      cas: momentPropTypes.momentDurationObj,
      dokonceno: PropTypes.bool,
      startCislo: PropTypes.number,
      onEdit: PropTypes.func.isRequired,
      onRemove: PropTypes.func.isRequired
    }).isRequired
  ).isRequired,
  typ: PropTypes.string.isRequired
};

export default Mezicasy;
