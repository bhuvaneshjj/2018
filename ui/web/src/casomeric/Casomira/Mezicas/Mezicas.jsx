import React from 'react';
import PropTypes from 'prop-types';
import momentPropTypes from 'react-moment-proptypes';
import { Button, Glyphicon } from 'react-bootstrap';
import { convertDuration } from '../../../Util';
import DroppableStartovniCislo from './DroppableStartovniCislo';
import './Mezicas.css';

const Mezicas = ({
  poradi,
  cas,
  dokonceno,
  startCislo,
  typ,
  canDrop,
  onDrop,
  onRemove,
  onEdit
}) => {
  const duration = convertDuration(cas);

  return (
    <React.Fragment>
      <div className="Mezicasy__cell Mezicas__poradi">{poradi}.</div>
      <div className="Mezicasy__leftCell Mezicas__cas">
        {cas && `${duration.hours}:${duration.mins}:${duration.secs},${duration.subsecs}`}
      </div>
      <DroppableStartovniCislo
        cas={cas}
        classNames="Mezicasy__middleCell Mezicas__startCislo"
        dokonceno={dokonceno}
        startCislo={startCislo}
        typ={typ}
        canDrop={canDrop}
        onDrop={onDrop}
      />
      <div className="Mezicasy__cell Mezicas__buttons">
        <Button bsSize="small" className="Mezicas__button--remove" onClick={onRemove}>
          <Glyphicon glyph="remove" />
        </Button>
        <Button bsSize="small" className="Mezicas__button--edit" onClick={onEdit} disabled={true}>
          <Glyphicon glyph="edit" />
        </Button>
      </div>
    </React.Fragment>
  );
};

Mezicas.propTypes = {
  cas: momentPropTypes.momentDurationObj,
  dokonceno: PropTypes.bool,
  poradi: PropTypes.number.isRequired,
  startCislo: PropTypes.number,
  typ: PropTypes.string.isRequired,
  canDrop: PropTypes.func.isRequired,
  onDrop: PropTypes.func.isRequired,
  onEdit: PropTypes.func.isRequired,
  onRemove: PropTypes.func.isRequired
};

Mezicas.defaultProps = {
  cas: undefined,
  dokonceno: undefined,
  startCislo: undefined
};

export default Mezicas;
