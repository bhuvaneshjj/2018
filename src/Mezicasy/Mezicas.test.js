import React from 'react';
import renderer from 'react-test-renderer';
import { Table } from 'react-bootstrap';
import moment from 'moment';
import Mezicas from './Mezicas';

const fakeOnClick = () => {};

it('jeden mezičas', () => {
  const component = renderer.create(
    <Table>
      <tbody>
        <Mezicas poradi={1} duration={moment.duration(34536)} onClick={fakeOnClick} />
      </tbody>
    </Table>
  );
  expect(component.toJSON()).toMatchSnapshot();
});

it('tři mezičasy', () => {
  const component = renderer.create(
    <Table>
      <tbody>
        <Mezicas poradi={1} duration={moment.duration(34536)} onClick={fakeOnClick} />
        <Mezicas poradi={2} duration={moment.duration(3453678)} onClick={fakeOnClick} />
        <Mezicas poradi={3} duration={moment.duration(345369874)} onClick={fakeOnClick} />
      </tbody>
    </Table>
  );
  expect(component.toJSON()).toMatchSnapshot();
});
