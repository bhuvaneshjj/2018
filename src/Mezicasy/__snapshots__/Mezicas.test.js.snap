// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`jeden mezičas 1`] = `
<table
  className="table"
>
  <tbody>
    <tr>
      <td>
        1
        .
      </td>
      <td>
        0
        :
        00
        :
        34
        ,
        53
      </td>
      <td>
        
      </td>
      <td>
        <button
          className="btn btn-xs btn-danger"
          disabled={false}
          onClick={[Function]}
          type="button"
        >
          x
        </button>
      </td>
    </tr>
  </tbody>
</table>
`;

exports[`tři mezičasy 1`] = `
<table
  className="table"
>
  <tbody>
    <tr>
      <td>
        1
        .
      </td>
      <td>
        0
        :
        00
        :
        34
        ,
        53
      </td>
      <td>
        
      </td>
      <td>
        <button
          className="btn btn-xs btn-danger"
          disabled={false}
          onClick={[Function]}
          type="button"
        >
          x
        </button>
      </td>
    </tr>
    <tr>
      <td>
        2
        .
      </td>
      <td>
        0
        :
        57
        :
        33
        ,
        67
      </td>
      <td>
        
      </td>
      <td>
        <button
          className="btn btn-xs btn-danger"
          disabled={false}
          onClick={[Function]}
          type="button"
        >
          x
        </button>
      </td>
    </tr>
    <tr>
      <td>
        3
        .
      </td>
      <td>
        23
        :
        56
        :
        09
        ,
        87
      </td>
      <td>
        
      </td>
      <td>
        <button
          className="btn btn-xs btn-danger"
          disabled={false}
          onClick={[Function]}
          type="button"
        >
          x
        </button>
      </td>
    </tr>
  </tbody>
</table>
`;
